x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/configs/index.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiCodeBlock,
+  EuiSpacer,
+  EuiSkeletonText,
+  EuiCallOut,
+  EuiLink,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import type { PackageInfo } from '../../../../../types';
+
+import { sendGetInputsTemplates, useStartServices } from '../../../../../hooks';
+
+interface ConfigsProps {
+  packageInfo: PackageInfo;
+}
+
+export const Configs: React.FC<ConfigsProps> = ({ packageInfo }) => {
+  const [configs, setConfigs] = useState<string | undefined>(undefined);
+  const [error, setError] = useState<undefined | Error>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const { notifications } = useStartServices();
+  const pkgName = packageInfo.name;
+  const pkgVersion = packageInfo.version;
+  // @ts-ignore-line
+  const notInstalled = !packageInfo?.installationInfo;
+
+  useEffect(() => {
```

Comment:
we should probably use `useRequest` hooks here instead of an effect

Reply:
In my first test I was trying with [useRequest](https://github.com/elastic/kibana/pull/168827/commits/7ca752d373d7973fc8d0501c5f4ae1305b662a8d#diff-7ac5e79652fc158965da85ac26327695852d38f8f9328438b75e8b1e0d1344a8R316-R329), but the endpoint is so slow that something weird was happening with the react state and didn't sync up correctly. I introduced the `useEffect` and it's now working fine.

Reply:
Sorry I was thinking about the `useQuery` hook not `useRequest`

something like this should work I think, and will be more robust than `useEffect` 

```
const { data: inputTemplateResponse } = useQuery(
    ['getInputsTemplates', pkgName, pkgVersion],
    () => sendGetInputsTemplates({ pkgName, pkgVersion }, { format: 'yml' })
  );
  ```

Reply:
Can we actually try using a `react-query` based hook here? Most of the other calls in integrations should be using those hooks now. 

```ts
// In `hooks/use_request/epm.ts`
export function useGetInputsTemplatesQuery(
  { pkgName, pkgVersion }: GetInputsTemplatesRequest["params"],
  query: GetInputsTemplatesRequest["query"],
) {
  return useQuery<GetInputsTemplatesResponse, RequestError>(
    ["inputsTemplates", pkgName, pkgVersion, query],
    () =>
      sendRequestForRq<GetInputsTemplatesResponse>({
        path: epmRouteService.getInputsTemplatesPath(pkgName, pkgVersion),
        method: "get",
        query,
        version: API_VERSIONS.public.v1, // side note could we move this up to `sendRequestForRq` so it's not duplicated everywhere?
      }),
  );
}


const { data: configs } = useGetInputTemplatesQuery({ pkgName, pkgVersion }, { format: 'yaml' })
```

> EDIT: the issue was that changing from one integration to another, the configs from the previous integration where shown for a couple of seconds, until the new config was retrieved. I tried it all to avoid it and useEffect did the trick

Sounds like we probably need to reset the loading state when `pkgName` and `pkgVersion` change. RQ should do this for us and we can grab `isLoading` off of the return value. Also, since `packageInfo` comes in via props we shouldn't need to copy it to state or track changes in an effect.

Reply:
Solved! There was a misterious sync up issue that was probably due to my local acting up